test/dynamo/test_misc.py

```diff
@@ -3819,6 +3819,67 @@ def fn(x, y):
         res = opt_fn(x, y)
         self.assertTrue(same(ref, res))
 
+    def test_int_neg(self):
+        def int_neg(a, b):
+            x = a.shape[0]
+            y = b.shape[0]
+            return -x * -y * a * b
+
+        torch._dynamo.testing.standard_test(self, int_neg, 2)
+
+    def test_has_attr_torch(self):
+        def fn_custom_get_attr(x, y):
+            if hasattr(torch.ops.aten, "mul"):
+                x = x * y
+                return x
+            else:
+                return x
+
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.Unsupported,
+            "hasattr with custom __getattr__",
+        ):
+            torch._dynamo.testing.standard_test(self, fn_custom_get_attr, 2)
+
+        def fn_safe(x, y):
+            if hasattr(torch.version, "git_version"):
+                x = x * y
+                return x
+            else:
+                return x
+
+        torch._dynamo.testing.standard_test(self, fn_safe, 2)
+
+    def test_has_attr_func(self):
+        def outer_custom_fn(x, y):
+            return x * y
+
+        outer_custom_fn.fake_attr = 3
+
+        def getattr_func(x, y):
+            z = outer_custom_fn(x, y)
+            if hasattr(outer_custom_fn, "fake_attr"):
+                return -z
+            return z
+
+        torch._dynamo.testing.standard_test(self, getattr_func, 2)
+
+    def test_has_attr_module(self):
+        class TestModule(torch.nn.Module):
+            def forward(self, x, y):
+                return x * y
+
+        tm = TestModule()
+        tm.some_val = -1
+
+        def getattr_func(x, y):
+            z = tm.forward(x, y)
+            if hasattr(tm, "fake_attr"):
+                return z * tm.some_val
+            return z
+
+        torch._dynamo.testing.standard_test(self, getattr_func, 2)
```

Comment:
`nn.Module`s have fake attributes (parameters, modules, and buffers) that will be missed by getattr_static since they are not stored on the module's `__dict__`.  We need to workaround on the custom for attributes on modules.

Reply:
NNModuleVariable has its own `var_getattr` and `call_hasattr` thought, unchanged by this PR?